src/main/java/org/opensearch/knn/index/codec/KNN990Codec/NativeEngineFieldVectorsWriter.java

```diff
@@ -54,7 +54,7 @@ static NativeEngineFieldVectorsWriter<?> create(final FieldInfo fieldInfo, final
         throw new IllegalStateException("Unsupported Vector encoding : " + fieldInfo.getVectorEncoding());
     }
 
-    NativeEngineFieldVectorsWriter(final FieldInfo fieldInfo, final InfoStream infoStream) {
+    private NativeEngineFieldVectorsWriter(final FieldInfo fieldInfo, final InfoStream infoStream) {
```

Comment:
making this change resolves this comment: https://github.com/opensearch-project/k-NN/pull/1855#discussion_r1692067872